File: scraper/db/team.py
"""Résolution de l'équipe (tenant) à partir d'un utilisateur.

Depuis la Partie 4 du schéma, le tenant réel est `teams` : plusieurs
utilisateurs peuvent partager les mêmes prospects/campagnes/SMTP via
`team_members`. `sender_profiles`/`accounts` sont désormais scopés par
`team_id`, pas par `user_id`.
"""

from __future__ import annotations

from typing import Any


def resoudre_team_id(client: Any, user_id: str) -> str | None:
    """Retourne le team_id de l'utilisateur, ou None s'il n'appartient à aucune équipe."""
    resp = (
        client.table("team_members")
        .select("team_id")
        .eq("user_id", user_id)
        .limit(1)
        .maybe_single()
        .execute()
    )
    data = resp.data if resp else None
    return data.get("team_id") if data else None

File: frontend/src/lib/queries/team.ts
"use client";

import { useQuery } from "@tanstack/react-query";

import { createClient } from "@/lib/supabase/client";
import type { Team, TeamRole } from "@/types/database";

export const CURRENT_TEAM_KEY = ["current_team"] as const;

export interface CurrentTeam {
  userId: string;
  teamId: string;
  role: TeamRole;
  team: Team;
}

// L'équipe est désormais le tenant réel (plusieurs utilisateurs peuvent
// partager les mêmes prospects/campagnes/SMTP). En v1, un utilisateur
// n'appartient qu'à une seule équipe — pas de sélecteur d'équipe dans l'UI
// pour l'instant, mais le modèle de données (team_members) supporte déjà
// plusieurs équipes par utilisateur si besoin plus tard.
export function useCurrentTeam() {
  const supabase = createClient();

  return useQuery({
    queryKey: CURRENT_TEAM_KEY,
    queryFn: async (): Promise<CurrentTeam | null> => {
      const {
        data: { user },
      } = await supabase.auth.getUser();
      if (!user) return null;

      const { data, error } = await supabase
        .from("team_members")
        .select("team_id, role, team:teams(*)")
        .eq("user_id", user.id)
        .limit(1)
        .maybeSingle();

      if (error) throw error;
      if (!data) return null;

      const row = data as unknown as { team_id: string; role: TeamRole; team: Team };

      return {
        userId: user.id,
        teamId: row.team_id,
        role: row.role,
        team: row.team,
      };
    },
  });
}
